### DISCUSSION TITLE
[supa_audit] adding the current user to audit records

### DISCUSSION THREAD

**Discussion by koenhandekyn** [supabase/supabase#18354] [Feature Requests]
this question is restricted to supa_audit. currently i'm using PaperTrail (gem) in ruby on rails applications which offers a very similar functionality. 

one of the key differences it seems is that there we're able to log also the current user, which is kind of key in many auditing contexts.

has this been on the table? some thoughts about it? how could we get other contextual data into the auditing sytem?

**Comment by tatupesonen** [comment]
What you could do is set the user on the DB session, for instance:
```sql
SET yourapp.db_session_user = 'username';
```

And then use `current_setting('yourapp.db_session_user')` to get the value in queries.
